flink-runtime/src/test/java/org/apache/flink/runtime/metrics/util/MetricUtilsTest.java

```diff
@@ -83,16 +81,24 @@ public void testStartMetricActorSystemRespectsThreadPriority() throws Exception
         final RpcService rpcService =
                 MetricUtils.startRemoteMetricsRpcService(
                         configuration, "localhost", RpcSystem.load());
-        assertThat(rpcService, instanceOf(AkkaRpcService.class));
-
-        final ActorSystem actorSystem = ((AkkaRpcService) rpcService).getActorSystem();
 
         try {
+            // dirty reflection code to avoid ClassCastExceptions
+            final Method getActorSystem = rpcService.getClass().getMethod("getActorSystem");
```

Comment:
Here it is: my archenemy reflection. Why can't we use a provided dependency of Akka?

Btw I just realized with a second look that this is test only. Please update the commit message. Is there something like testProvided in maven?

Reply:
ClassCastExceptions because the actual loading happens in the plugin loader but references here would use the system loader.

Reply:
Yes, all good, since it's test-only.